codalab/server/bundle_manager.py

```diff
@@ -522,18 +516,19 @@ def _construct_run_message(self, worker, bundle):
                 )
 
         # Figure out the resource requirements.
-        resources = message['resources'] = {}
-
-        resources['request_cpus'] = self._compute_request_cpus(bundle)
-        resources['request_gpus'] = self._compute_request_gpus(bundle)
-
-        resources['docker_image'] = self._get_docker_image(bundle)
-        resources['request_time'] = self._compute_request_time(bundle)
-        resources['request_memory'] = self._compute_request_memory(bundle)
-        resources['request_disk'] = self._compute_request_disk(bundle)
-        resources['request_network'] = bundle.metadata.request_network
-
-        return message
+        bundle_resources = self._compute_bundle_resources(bundle)
+        message['resources'] = bundle_resources.to_dict()
+
+    def _compute_bundle_resources(self, bundle):
+        return RunResources(
```

Comment:
We have an object that encapsulates all bundle resources rather than a dict that comes with serialization/deserialization methods. This is so that we don't shoot ourselves in the foot by missing a dict field and only noticing when things go wrong etc.